File: src/main/java/pl/mjedynak/idea/plugins/builder/gui/helper/GuiHelper.java
package pl.mjedynak.idea.plugins.builder.gui.helper;

import com.intellij.codeInsight.CodeInsightUtil;
import com.intellij.openapi.fileEditor.ex.IdeDocumentHistory;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiFile;
import javax.swing.Icon;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class GuiHelper {

    private GuiHelper() {
        throw new UnsupportedOperationException("Utility class");
    }

    public static void showMessageDialog(
            @Nullable Project project, @NotNull String message, @NotNull String title, @Nullable Icon icon) {
        Messages.showMessageDialog(project, message, title, icon);
    }

    public static void includeCurrentPlaceAsChangePlace(@NotNull Project project) {
        IdeDocumentHistory.getInstance(project).includeCurrentPlaceAsChangePlace();
    }

    public static void positionCursor(
            @NotNull Project project, @NotNull PsiFile psiFile, @NotNull PsiElement psiElement) {
        CodeInsightUtil.positionCursor(project, psiFile, psiElement);
    }
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/gui/helper/GuiHelper.kt
package io.github.kingg22.kobra.builder.gui.helper

import com.intellij.codeInsight.CodeInsightUtil
import com.intellij.openapi.fileEditor.ex.IdeDocumentHistory
import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.Messages
import com.intellij.psi.PsiElement
import com.intellij.psi.PsiFile
import javax.swing.Icon

public object GuiHelper {
    @JvmStatic
    public fun showMessageDialog(project: Project?, message: String?, title: String, icon: Icon?) {
        Messages.showMessageDialog(project, message, title, icon)
    }

    @JvmStatic
    public fun includeCurrentPlaceAsChangePlace(project: Project) {
        IdeDocumentHistory.getInstance(project).includeCurrentPlaceAsChangePlace()
    }

    @JvmStatic
    public fun positionCursor(project: Project, psiFile: PsiFile, psiElement: PsiElement) {
        CodeInsightUtil.positionCursor(project, psiFile, psiElement)
    }
}
